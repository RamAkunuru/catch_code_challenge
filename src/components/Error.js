import React from 'react';

export default ({ errorMessage }) => (
    <div className="error">
        { errorMessage }
    </div>
);
