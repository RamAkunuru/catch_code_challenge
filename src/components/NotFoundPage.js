import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

export default () => (
    <Fragment>
    404 - <Link to="/">GO home</Link>
</Fragment>
);
