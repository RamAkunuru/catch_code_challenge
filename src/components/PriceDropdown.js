/* eslint-disable jsx-a11y/no-onchange */
import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as productsActions from '../actionCreators';

const PriceDropdown = props => {
    const handlePriceDropdown = event => {
        const sortOrder = event.target.value === 'highest' ? 'desc' : 'asc';
        props.actions.sortByPrice(sortOrder);
    }

    return (
        <select className="price-dropdown" onChange={handlePriceDropdown}>
            <option value="highest">Highest Price</option>
            <option value="lowest">Lowest Price</option>
        </select>
    )
}

const mapDispatchToProps = dispatch => ({
    actions: bindActionCreators(productsActions, dispatch)
});

export { PriceDropdown }

export default connect(null, mapDispatchToProps)(PriceDropdown);