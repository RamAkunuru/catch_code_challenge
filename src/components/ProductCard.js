import React from 'react';

export default (props) => {
    const { name, salePrice, retailPrice, imageUrl, quantityAvailable } = props.product;

    return (
        <div className="product-card">
            <a href="#" title={name}>
                <div className="product-image">
                    <img src={imageUrl} alt={name}/>
                    <span>{quantityAvailable ? null : "SOLD OUT"}</span>
                </div>
                
                <div className="product-details">
                    <h2 className="product-name">{name}</h2>
                    <p className="product-price">{ retailPrice ? <del className="price">{retailPrice}</del> : "" }</p>
                    <p className="price sale-price">{salePrice}</p>
                </div>
            </a>
        </div>
    );
}