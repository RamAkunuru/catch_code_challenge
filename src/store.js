import { createStore, compose, applyMiddleware } from 'redux';
// import thunk from 'redux-thunk';
import reducer from './reducers';
import createSagaMiddleware from 'redux-saga';
import { watchGetProductDetails } from './sagas/saga';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

// Using thunks
// const store = createStore(reducer, composeEnhancers(
//     applyMiddleware(thunk)
// ));

const store = createStore(reducer, composeEnhancers(
    applyMiddleware(sagaMiddleware)
));

sagaMiddleware.run(watchGetProductDetails);

export default store;
