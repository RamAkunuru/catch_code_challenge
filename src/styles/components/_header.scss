.header {
  display: flex;
  justify-content: space-between;
  margin-bottom: 1rem;
  padding: 1rem 0;
}
