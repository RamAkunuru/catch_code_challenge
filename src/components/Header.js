import React from 'react';
import Logo from './Logo';
import PriceDropdown from './PriceDropdown';

export default () => (
    <div className="header">
        <h1><Logo width="13.2rem" /></h1>
        <PriceDropdown />
    </div>
)
