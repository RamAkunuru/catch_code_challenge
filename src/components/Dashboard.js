import React, { useEffect } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as productsActions from '../actionCreators';
import ProductCard from './ProductCard';

const Dashboard = (props) => {
    useEffect(() => {
        props.actions.getAPIDetails();
    }, []);

    const renderProducts = () => {
        const { products } = props;

        return products.map(product => <ProductCard product={product} key={product.id}/>);
    };

    return (
        <div className="dashboard">{props.products.length ? renderProducts() : null}</div>
    );
}

const mapStateToProps = state => ({
    products: state.productList
});

const mapDispatchToProps = dispatch => ({
    actions: bindActionCreators(productsActions, dispatch)
});

export { Dashboard }

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
