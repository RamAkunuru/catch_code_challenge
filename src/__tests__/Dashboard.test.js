import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../components/Dashboard';
import store from '../store';

const products = [
    { 
        id: "123",
        name: "abc",
        imageUrl: "https://google.com/something1",
        salePrice: 123, 
        retailPrice: 456, 
        quantityAvailable: 12
    }, 
    {
        id: "124",
        name: "def",
        imageUrl: "https://google.com/something2",
        salePrice: 124, 
        retailPrice: 457, 
        quantityAvailable: 14
    }
];

describe("Dashboard component", () => {
    test("render matches the snapshot", () => {
        const component = shallow(<Dashboard products={products} store={store} />);
        expect(component).toMatchSnapshot();
    });
});
