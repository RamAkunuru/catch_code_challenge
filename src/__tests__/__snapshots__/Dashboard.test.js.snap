// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component render matches the snapshot 1`] = `
<div
  className="dashboard"
>
  <_default
    key="123"
    product={
      Object {
        "id": "123",
        "imageUrl": "https://google.com/something1",
        "name": "abc",
        "quantityAvailable": 12,
        "retailPrice": 456,
        "salePrice": 123,
      }
    }
  />
  <_default
    key="124"
    product={
      Object {
        "id": "124",
        "imageUrl": "https://google.com/something2",
        "name": "def",
        "quantityAvailable": 14,
        "retailPrice": 457,
        "salePrice": 124,
      }
    }
  />
</div>
`;
